RestaurantList.js

```diff
@@ -3,33 +3,58 @@ import Restaurant from './Restaurant'
 import RestaurantModel from './RestaurantModel'
 
 class RestaurantList extends React.Component {
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
     this.displayName = "RestaurantList"
-
     this.state = {
-      restaurants : [
-        new RestaurantModel("Moses", "Burger", 2)
-        ,new RestaurantModel("המזנון", "Pita", 3)
-        ,new RestaurantModel("Fresh", "Bari", 1)
-        ,new RestaurantModel("Hachim", "Mideterian", 3)
-      ]
+      selectedRestaurant : 0
     }
   }
 
+  handleSelectedRestaurant(restaurant, i) {
+    console.log('selected restaurant ' + i)
+    this.setState({ selectedRestaurant: i });
+  }
+
   render() {
+    console.log(this.props.restaurants[this.state.selectedRestaurant].address)
     return (
       <div className="row">
         <div className="col-md-7">
-          <ul class="media-list">
-            {this.state.restaurants.map(function(restaurant, i) {
-              return <Restaurant model={restaurant} key={i} />
-            })}
+          <ul className="media-list">
+            {this.props.restaurants.map(
+              function(restaurant, i) {
+                var selected = (this.state.selectedRestaurant == i)
```

Comment:
I could, I actually learned about `=>` keeping the `this` reference automatically only later yesterday (after writing this code). What I'm doing here is passing this as the second parameter of map(fn, this). Another work around that worked for me was doing bind on the `function(restaurant, i)` function. Complex code.. not something I'm proud of. 

As for your comment, I think fat arrow is a better practice, I'm refactoring this too.